e2e-tests/src/staking.rs

```diff
@@ -97,3 +100,107 @@ pub fn wait_for_era_completion(
+    assert_eq!(
+        locked_stash_balance_before_payout.len(),
+        1,
+        "Expected locked balances for account {} to have exactly one entry!",
+        stash_account
+    );
+    payout_stakers(node, stash.clone(), era - 1);
+    let locked_stash_balance_after_payout = locks(&connection, &stash);
+    assert!(
+        locked_stash_balance_after_payout.is_some(),
+        "Expected non-empty locked balances for account {}!",
+        stash_account
+    );
+    let locked_stash_balance_after_payout = locked_stash_balance_after_payout.unwrap();
+    assert_eq!(
+        locked_stash_balance_after_payout.len(),
+        1,
+        "Expected non-empty locked balances for account to have exactly one entry {}!",
+        stash_account
+    );
+    assert!(locked_stash_balance_after_payout[0].amount > locked_stash_balance_before_payout[0].amount,
+            "Expected payout to be non zero in locked balance for account {}. Balance before: {}, balance after: {}",
+            stash_account, locked_stash_balance_before_payout[0].amount, locked_stash_balance_after_payout[0].amount);
+}
+
+pub fn batch_bond(
+    connection: &Connection,
+    stash_controller_key_pairs: &[(&KeyPair, &KeyPair)],
+    bond_value: u128,
+    reward_destination: RewardDestination<GenericAddress>,
+) {
+    let batch_bond_calls: Vec<_> = stash_controller_key_pairs
+        .iter()
+        .map(|(stash_key, controller_key)| {
+            let bond_call = compose_call!(
+                connection.metadata,
+                "Staking",
+                "bond",
+                GenericAddress::Id(AccountId::from(controller_key.public())),
+                Compact(bond_value),
+                reward_destination.clone()
+            );
+            compose_call!(
+                connection.metadata,
+                "Sudo",
+                "sudo_as",
+                GenericAddress::Id(AccountId::from(stash_key.public())),
+                bond_call
+            )
+        })
+        .collect::<Vec<_>>();
+
+    let xt = compose_extrinsic!(connection, "Utility", "batch", batch_bond_calls);
+    send_xt(
+        connection,
+        xt.hex_encode(),
+        "batch of bond calls",
+        XtStatus::InBlock,
+    );
+}
+
+pub fn batch_nominate(connection: &Connection, nominator_nominee_pairs: &[(&KeyPair, &KeyPair)]) {
+    let batch_nominate_calls: Vec<_> = nominator_nominee_pairs
+        .iter()
+        .map(|(nominator, nominee)| {
+            let nominate_call = compose_call!(
+                connection.metadata,
+                "Staking",
+                "nominate",
+                vec![GenericAddress::Id(AccountId::from(nominee.public()))]
+            );
+            compose_call!(
+                connection.metadata,
+                "Sudo",
+                "sudo_as",
+                GenericAddress::Id(AccountId::from(nominator.public())),
+                nominate_call
+            )
```

Comment:
do I understand correctly that you use `sudo_as` because `connection` is already signed by `sudo`? if so, then if I remember correctly, actually, you can call `set_signer` on `connection` to change the signer; however, you need to be aware of its kinda inconvenient signature (`&mut self`), so probably cloning `connection` beforehand is the easiest way to tackle this